src/mixins/embed-button.js

```diff
@@ -72,7 +88,25 @@ export default {
       );
       this.$root.$emit('openPopupNoticeOverlay', w);
     },
-
+    setIsDisplayed() {
+      this.isDisplayed = true;
+    },
+    async handleReadEvent() {
+      if (!this.isRead && this.isReadTimerEnded && this.isDisplayed) {
+        await this.postReadEvent();
+      }
+    },
+    async postReadEvent() {
+      if (this.isRead) return;
+      await apiPostLikeButtonReadEvent(
```

Comment:
Since `this.isRead = true` is after `await`, it seems possible to trigger `apiPostLikeButtonReadEvent` multiple times?

Reply:
Also, I'm not sure about the responsibilities of `postReadEvent` and `handleReadEvent`, it looks weird to me.
I originally thought `handleReadEvent` is a checker which is triggered every time when any of the variables in the condition changed, to see if the whole condition holds for calling `postReadEvent`.
In this case, `postReadEvent` can only be called by `handleReadEvent`.
`handleReadEvent` could be called multiple times, but it will check and set `isRead` before `postReadEvent`, to ensure that `postReadEvent` will be called only once.
But the current situation is that `postReadEvent` will be called directly without `handleReadEvent`, so it seems weird to me.

Reply:
If I put it before awai then it cannot recover if it throws
not sure if its relevant

Reply:
Simply put I just want a handler for handling displayed + timer relationship, then another one for directly triggering post event